src/test/java/uk/gov/hmcts/reform/unspec/service/CoreCaseDataServiceTest.java

```diff
@@ -65,43 +81,66 @@ void init() {
+            .build();
+        private final CaseDetails caseDetails = CaseDetailsBuilder.builder()
+            .data(caseData)
+            .build();
 
         @BeforeEach
         void setUp() {
             when(idamClient.getUserInfo(USER_AUTH_TOKEN)).thenReturn(UserInfo.builder().uid(USER_ID).build());
 
             when(coreCaseDataApi.startEventForCaseWorker(USER_AUTH_TOKEN, SERVICE_AUTH_TOKEN, USER_ID, JURISDICTION,
-                CASE_TYPE, Long.toString(CASE_ID), EVENT_ID))
-                .thenReturn(buildStartEventResponse());
+                                                         CASE_TYPE, Long.toString(CASE_ID), EVENT_ID
+            )).thenReturn(buildStartEventResponse());
+
+            when(coreCaseDataApi.submitEventForCaseWorker(
+                USER_AUTH_TOKEN,
+                SERVICE_AUTH_TOKEN,
+                USER_ID,
+                JURISDICTION,
+                CASE_TYPE,
+                Long.toString(CASE_ID),
+                true,
+                buildCaseDataContent()
+                 )
+            ).thenReturn(caseDetails);
         }
 
         @Test
         void shouldStartAndSubmitEvent_WhenCalled() {
             service.triggerEvent(CASE_ID, CaseEvent.valueOf(EVENT_ID));
 
             verify(coreCaseDataApi).startEventForCaseWorker(USER_AUTH_TOKEN, SERVICE_AUTH_TOKEN, USER_ID,
-                JURISDICTION, CASE_TYPE, Long.toString(CASE_ID), EVENT_ID);
-            verify(coreCaseDataApi).submitEventForCaseWorker(USER_AUTH_TOKEN, SERVICE_AUTH_TOKEN, USER_ID, JURISDICTION,
-                CASE_TYPE, Long.toString(CASE_ID), true, buildCaseDataContent());
+                                                            JURISDICTION, CASE_TYPE, Long.toString(CASE_ID), EVENT_ID
+            );
+            verify(coreCaseDataApi).submitEventForCaseWorker(
+                USER_AUTH_TOKEN,
+                SERVICE_AUTH_TOKEN,
+                USER_ID,
+                JURISDICTION,
+                CASE_TYPE,
+                Long.toString(CASE_ID),
+                true,
+                buildCaseDataContent()
+            );
         }
 
         private StartEventResponse buildStartEventResponse() {
             return StartEventResponse.builder()
                 .eventId(EVENT_ID)
                 .token(EVENT_TOKEN)
-                .caseDetails(CaseDetails.builder()
-                    .data(Map.of("data", "some data"))
-                    .build())
+                .caseDetails(caseDetails)
                 .build();
         }
 
         private CaseDataContent buildCaseDataContent() {
             return CaseDataContent.builder()
                 .eventToken(EVENT_TOKEN)
-                .event(Event.builder()
-                    .id(EVENT_ID)
-                    .build())
-                .data(Map.of("data", "some data"))
+                .event(Event.builder().id(EVENT_ID).build())
+                .data(caseDetailsConverter.convertToMap(caseData))
```

Comment:
Do you think we should test extracted start and submit methods in separate tests?

Reply:
Technically yes, however they are one of change